### DISCUSSION TITLE
Feature Request: Ability to set timeout value in .NET library

### DISCUSSION THREAD

**Discussion by r-brown** [Labs64/NetLicensing-Community#78] [Ideas & Feature Requests]
Would it be possible for you to make an updated version of .NET library: https://github.com/Labs64/NetLicensingClient-csharp and an updated Nuget build, where you expose the timeout value on the http-request you send towards your backend service?
 
We would like to be able to handle timeouts, but the current timeout is too high for us and it would be great if we could set it through e.g. a constructor parameter.

**Comment by boskjoett** [comment]
I assume your .NET library is using the HttpClient class for HTTP requests. It has a default timeout of 100 seconds.
See [https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.timeout?view=net-5.0](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.timeout?view=net-5.0)
It would be nice to be able to adjust this timeout.
This timeout is only seen when your backend does not respond at all, which I hope is unlikely.
We tested our own license failover system by blocking access to your backend in our firewall and in this case it takes 100 seconds to detect that your backend is unreachable.

**Comment by r-brown** [comment]
Latest NetLicensing Client (C#) release [v2.5.1](https://www.nuget.org/packages/NetLicensingClient-csharp/) introduced timeout parameter for the `validate()` method.

New package is also available @ [nuget](https://www.nuget.org/packages/NetLicensingClient-csharp/)
